### ISSUE TITLE
Preference configuration through managed storage manifest

### ISSUE THREAD

**Issue by BV-Trance** [stoically/temporary-containers#303]
Hi Stoically,

I am attempting to push out Firefox ESR 68.0.2 with the Temporary-Containers extension using group policy, which I have successfully done. However, I would like to know if there is any way to enable the "Automatic Mode" by default as we don't want our users to interact with the Add-on manager in Firefox.

It seems if this feature is able to be enabled by default eg by pushing out a settings file or by other means, it would fulfill our needs.

Thanks in advance! Your extension is awesome :)

BV

**Comment by stoically** [comment]
Hey, thanks, glad you like it!

Currently that's unfortunately not possible other than by copying the extension and it's pre-configured storage to the appropriate locations, see #216 for details.

Though, I did some research, and it seems the way to go is to read `data` set in the [native manifest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests) and use that to override preferences - [that's how uBlock Origin does it too](https://github.com/gorhill/uBlock/wiki/Deploying-uBlock-Origin). I'll look into implementing that.

**Comment by BV-Trance** [comment]
Thank you for your reply.

I will look into your ideas and give them a go. I appreciate you pointing me in the right direction and I look forward to seeing how this extension evolves over time.

**Comment by stoically** [comment]
Support for managed storage is now available in [v1.4beta1](https://github.com/stoically/temporary-containers/releases). Preferences from the managed storage overwrite local preferences when the Add-on initializes.

To add managed storage, create a file called `.json` in one of the [supported locations](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#Manifest_location) in the [needed format](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#Managed_storage_manifests). The `data` key must contain a full preferences export.

Example:

`~/.mozilla/managed-storage/{c607c8df-14a7-4f28-894f-29e8722976af}.json`

with content

```json
{
  "name": "{c607c8df-14a7-4f28-894f-29e8722976af}",
  "description": "ignored",
  "type": "storage",
  "data":
  {
    "version": "1.4beta1",
    "date": 1572695327640,
    "preferences": {
      "automaticMode": {
        "active": true,
        "newTab": "created"
      },
      ...<rest of the export>...
    }
  }
}
```

[Also added the comment to the wiki](https://github.com/stoically/temporary-containers/wiki/Managed-Storage). Hope that helps. Let me know if something doesn't work as expected.
